ci/Jenkinsfile

```diff
@@ -259,19 +269,30 @@ pipeline {
             }
         }
         stage( '5. FINALIZE' ) {
-            when { 
-                expression {
-                   STATUS == 'Passed'
-                }
-            }
-            agent { label "${machine}-emc" }
+            agent { label NodeName[machine].toLowerCase() }
             steps {
                 script {
-                    try {
-                        sh(script: """${GH} pr edit ${env.CHANGE_ID} --repo ${repo_url} --remove-label "CI-${Machine}-Running" --remove-label "CI-${Machine}-Building" --add-label "CI-${Machine}-${STATUS}" """, returnStatus: true)
-                        sh(script: """${GH} pr comment ${env.CHANGE_ID} --repo ${repo_url} --body "**CI ${STATUS}** ${Machine} at <br>Built and ran in directory \\`${HOME}\\`" """, returnStatus: true)
-                    } catch (Exception e) {
-                        echo "Failed to update label from Running to ${STATUS}: ${e.getMessage()}"
+                    sh(script: """
+                        labels=\$(gh pr view ${env.CHANGE_ID} --repo ${repo_url} --json labels --jq '.labels[].name')
+                        for label in \$labels; do
+                            if [[ "\$label" == *"${Machine}"* ]]; then
+                                gh pr edit ${env.CHANGE_ID} --repo ${repo_url} --remove-label "\$label"
+                            fi
+                        done
+                    """, returnStatus: true)
+                    sh(script: """${GH} pr edit ${env.CHANGE_ID} --repo ${repo_url} --add-label "CI-${Machine}-${STATUS}" """, returnStatus: true)
+                    if (fileExists("${HOME}/RUNTESTS/ci-run_check.log")) {
+                        sh(script: """echo "**CI ${STATUS}** ${Machine} at <br>Built and ran in directory \\`${HOME}\\`\n\\`\\`\\`\n" | cat - ${HOME}/RUNTESTS/ci-run_check.log > temp && mv temp ${HOME}/RUNTESTS/ci-run_check.log""", returnStatus: true)
+                        sh(script: """${GH} pr comment ${env.CHANGE_ID} --repo ${repo_url} --body-file ${HOME}/RUNTESTS/ci-run_check.log  """, returnStatus: true)
+                    }
+                    if (STATUS == 'Passed') {
+                        try {
+                            sh(script: "rm -Rf ${HOME}/*")
```

Comment:
This makes me pucker up a bit. I'm assuming this is the Jenkins variable and not the shell variable, but I still think we should rename the variable to avoid the conflict with the shell `$HOME`.

Reply:
Oh my absolutely should be disambiguated!!!  Changing it to **CUSTOME_WORKSPACE** really enhances the readability too.